File: 프로그래머스/unrated/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.java
import java.util.stream.IntStream;

class Solution {
    public int solution(int n) {
        int result = 0;
		
		if(n % 2 == 0) {
			result = IntStream.rangeClosed(1, n).filter(num -> num % 2 == 0).map(num -> (int) Math.pow(num, 2)).sum();
		} else {
			result = IntStream.rangeClosed(1, n).filter(num -> num % 2 != 0).sum();
		}
		
        return result;
    }
}

File: 프로그래머스/0/181935. 홀짝에 따라 다른 값 반환하기/홀짝에 따라 다른 값 반환하기.py
def solution(n):
    answer = 0

    for num in range(n+1):
        if(n % 2):
            if(num % 2):
                answer += num
            else:
                continue                
        else:
            if(num % 2):
                continue                
            else:
                answer += num ** 2
    return answer